src/status.rs

```diff
@@ -638,3 +638,101 @@ impl StatusClass {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use super::StatusCode::*;
+
+    // Check that the following entities are properly inter-connected:
+    //   - numerical code
+    //   - status code
+    //   - default code (for the given status code)
+    fn validate(num:u16, status_code:StatusCode, default_code:StatusCode, reason:Option<&str>) {
+        let actual_code = StatusCode::from_u16(num);
+        if actual_code != status_code  {
```

Comment:
These can all be `assert_eq!`, and you'll get a decent error message automatically.